lib/pbench/test/unit/server/query_apis/commons.py

```diff
@@ -282,16 +291,22 @@ def test_http_exception(
         user_ok,
         find_template,
         pbench_token,
+        provide_metadata,
     ):
         """
         Check that an exception in calling Elasticsearch is reported correctly.
         """
         if not self.elastic_endpoint:
             pytest.skip("skipping " + self.test_http_exception.__name__)
 
-        index = self.build_index(
-            server_config, self.date_range(self.payload["start"], self.payload["end"])
-        )
+        if self.payload.get("start") and self.payload.get("end"):
+            index = self.build_index(
+                server_config,
+                self.date_range(self.payload["start"], self.payload["end"]),
+            )
+        else:
+            index = self.build_index_from_metadata(provide_metadata)
```

Comment:
I still don't get what you're accomplishing by passing the value of the fixture to your helper method. That's not a fixture, so referencing a fixture accomplishes nothing, and the code doesn't use it as a parameter. I don't think it should be here at all.

Reply:
Right:  as I understand it, referencing a PyTest "fixture" is only significant if it is referenced in the function signature of a "test" (i.e., a function whose name begins with `test_`) or another fixture; inside a test or fixture which references it in its signature, referencing a fixture produces the cached value which was originally returned (or `yield`ed) by the fixture.

Since `provide_metadata` doesn't yield or return a value, the reference in the cited code will produce `None` (or whatever value was produced by the last statement in the fixture), and it won't have any other effect.  So, there's no point in referencing it here.